build_tools/ci/cpu_comparison/input_generator.py

```diff
@@ -95,22 +93,43 @@ def get_numpy_type(element_type):
         raise ValueError("Invalid or unsupported element type: " + element_type)
 
 
+def verify_determinism():
+    """
+    Assert that the approach we use is deterministic across space and time...
+    we don't want OS, numpy version, etc, influencing random values.
+    """
+    seed = 1
+    rng = np.random.Generator(np.random.MT19937(np.random.SeedSequence(1)))
+    test_values = [x for x in rng.integers(0, 100000, 4)]
+    expected_test_values = [24067, 90095, 72958, 10894]
+    if test_values != expected_test_values:
+        message = (
+            "The approach for generating pseudo-random numbers does not appear to be "
+            "reproducible across platforms (OSs, numpy versions, etc.). The expected "
+            "pseudo-random values (generated on a different platform) were "
+            f"{expected_test_values}, but the values generated on this platform are "
+            f"{test_values}."
+        )
+        raise ValueError(message)
+
+
 def write_input(bin_filename, num_elements, element_type, input_number):
     # Random integer values in range [lower_bound, upper_bound)
     # will be generated for the input data.
     lower_bound = 0
     upper_bound = 10
 
-    # Fix the seed for each input, based on the input number.
-    np.random.seed(1 + input_number)
+    verify_determinism()
+    # We have now verified that approach below is deterministic across platforms.
+    seed = input_number
```

Comment:
This was something I was thinking about when debugging flaky tests last week (something == the seed). Do we print/report this number (the seed) somewhere in the test logs/outputs?

Reply:
Yea I guess I noticed that it is fixed but I think the "right" approach is to "free" it but log/report the seed.

Reply:
So pass the seed in as an argument somewhere? One seed for each operand? Like, generate the numpy operands "standalone"? 

That's kind of what happens here: https://github.com/nod-ai/iree-amd-aie/blob/9f809c6c27730894275919347f2c44ff40e4a505/build_tools/ci/cpu_comparison/run_test.sh#L294 

You still need to run that script to create the input arrays

Reply:
yea sure cool good enough - so then we can move to "actually random in CI" + "print the seed for reproducibility" ?

Reply:
We've agreed to disagree for now on using different seeds between CI runs, but I've changed the PR now and made it easier for a developer to set the seed in any way they want now. @makslevental can you accept this PR if you're ok with the compromise!

Reply:
discussed offline - tabled until future bikeshedding sessions.

Reply:
Wouldn't freeing the seed mean you can't tell whether you have a failure because your code change caused it or because it found a failing seed?  I would think that for CI, we'd want to prioritize detecting breaking code changes, while for nightly regressions, we'd want to do random seed testing.

Reply:
Yeah that's what I think, except maybe even nightly regressions should keep frozen seeds -- IMO if you think the coverage of a test isn't good enough, increase the size of the random tensor rather than freeing the seed. Or improve the random distribution you're using.